infra/docker-ops.sh

```diff
@@ -1,42 +1,112 @@
 
-# shellcheck disable=SC2210
-IMAGE_VERSION=$(cat apps/"${IMAGE_NAME}"/VERSION 2>/dev/null || echo "${2}")
-
-LOCAL_IMAGE="local/${IMAGE_NAME}:${IMAGE_VERSION}"
-
-# We want to push same images on multiple docker registries
+# Static vars
 DOCKER_REPOS=("public.ecr.aws/htec")
+BASE_DIR="$PWD/apps"
+FULL_PATH=$BASE_DIR/$REPOSITORY_NAME/$FLAVOR_NAME
+SUBDIRS=$(find ${FULL_PATH} -maxdepth 1 -type d | wc -l)
+IMAGE_VERSION=$(cat "${FULL_PATH}"/VERSION 2>/dev/null || echo "${3}")
+IMAGES=()
 
-build() {
-  docker build -t "${LOCAL_IMAGE}" --build-arg "VERSION=${IMAGE_VERSION}" "apps/${IMAGE_NAME}/"
+# Main function is responsible for checking if manadatory variable is provided
+# If provided then loop through the docker repos and build an image for every repo
+main(){
+  if [ -z "$REPOSITORY_NAME" ] ; then
+      echo -e "     
+      #####################################\n
+      Mandatory argument is not provided\n
+      #####################################
+      "
+    else
+      for REPO in "${DOCKER_REPOS[@]}"; do
+        build_and_prepare_data
+      done
+  fi
+  # Wait for prepared data from previous function and proceed with the push and cleanup process.
+  push_image $IMAGES $DRY_RUN
+  remove_image $IMAGES $DRY_RUN
 }
 
-push() {
-  if [ "${LOCAL_TEST}" == "true" ]; then
-      echo "Skipping 'docker push' for local testing."
-      return
-  fi
-  for REPO in "${DOCKER_REPOS[@]}"; do
-    NEW_IMAGE="$REPO/${IMAGE_NAME}:${IMAGE_VERSION}"
-    docker tag "local/${IMAGE_NAME}:${IMAGE_VERSION}" "${NEW_IMAGE}"
-    docker push "${NEW_IMAGE}"
-
-    # Remove tag after push
-    docker rmi "${NEW_IMAGE}"
-  done
+# TO-DO
+# Improve logic of the function build_and_prepare_data
+# This functions is responsible for formating the given input and bulding the docker images.
+build_and_prepare_data() {
+    if [[ "${SUBDIRS}" -eq 1 ]]
+    then
+      for FOLDER in $(ls -d ${FULL_PATH}); do
+        IMAGE_NAME=$REPO/${REPOSITORY_NAME}:${IMAGE_VERSION}
+        build_image $IMAGE_NAME $FOLDER $DRY_RUN
+        # Append value to a variable
+        IMAGES+="${IMAGE_NAME};"
+      done
+    else
+      for FOLDER in $(ls -d ${FULL_PATH}/*/); do
+        if [ -f "${FOLDER}/Dockerfile" ]; then
+          IMAGE_NAME=$REPO/${REPOSITORY_NAME}:${IMAGE_VERSION}-$(basename "$FOLDER/")
+          build_image $IMAGE_NAME $FOLDER $DRY_RUN
+          # Append value to a variable
+          IMAGES+="${IMAGE_NAME};"
+        else
+          FOLDER=$(ls -d ${FULL_PATH})
+          IMAGE_NAME=$REPO/${REPOSITORY_NAME}:${IMAGE_VERSION}
+          build_image $IMAGE_NAME $FOLDER $DRY_RUN
+          # Append value to a variable
+          IMAGES+="${IMAGE_NAME};"
+        fi
+      done
+    fi
 }
 
-cleanup() {
-  if [ "${LOCAL_TEST}" == "true" ]; then
-      echo "Skipping cleanup"
+build_image(){
+  local IMAGE_NAME="$1"
+  local FOLDER="$2"
+  local DRY_RUN="$3"
+
+  if [ "$DRY_RUN" = "true" ] ; then
+    docker build -t local/$IMAGE_NAME "$FOLDER/"
```

Comment:
Dry-run is all about logging. So print explicit message that says which image is built and why.